inc/wpml-compatibility-test-tools.class.php

```diff
@@ -177,6 +156,65 @@ private function translate_strings( $context, $languages, $template ) {
 		}
 	}
 
+	public function generate_strings_translations() {
+		check_ajax_referer( 'mt_generate_strings_translations', '_mt_mighty_nonce' );
+
+		$contexts  = isset( $_POST['contexts'] ) ? (array) $_POST['contexts'] : false;
+		$languages = isset( $_POST['languages'] ) ? $_POST['languages'] : false;
+		$template  = isset( $_POST['template'] ) ? $_POST['template'] : false;
+		$count     = isset( $_POST['count'] ) ? $_POST['count'] : false;
+		$offset    = isset( $_POST['offset'] ) ? $_POST['offset'] : 0;
+
+		if ( ! $contexts || ! $languages || ! $template ) {
+			echo 0;
+			wp_die();
+		}
+
+		// Strings batch threshold
+		$limit = 50;
+
+		global $wpdb;
+
+		$esc_contexts = array_map( function ( $context ) {
+			return "'" . esc_sql( $context ) . "'";
+		}, $contexts );
+		$esc_contexts = implode( ",", $esc_contexts );
+
+		// Skip count if process started.
+		if ( empty( $count ) ) {
+			$count = $wpdb->get_var( "SELECT COUNT(*) FROM {$wpdb->prefix}icl_strings WHERE context IN ({$esc_contexts})" );
+
+			// Update settings only on first run.
+			self::update_option( 'string_auto_translate_context', $contexts );
+			self::update_option( 'string_auto_translate_languages', $languages );
+			self::update_option( 'string_auto_translate_template', $template );
+		}
+
+		if ( $offset <= $count ) {
+			$strings = $wpdb->get_results( "SELECT id, language, context, value FROM {$wpdb->prefix}icl_strings WHERE context IN ({$esc_contexts}) LIMIT {$offset}, {$limit}" );
+			$this->translate_strings( $strings, $languages, $template );
+
+			// Update offset.
+			$offset += $limit;
+
+			// Calculate progress percentage.
+			$strings_left_count = max( $count - $offset, 0 );
+			$progress           = floor( 100 - $strings_left_count * 100 / $count );
+
+			// Response.
+			echo json_encode(

```

Comment:
you can use https://developer.wordpress.org/reference/functions/wp_send_json/ , it will send response and do `die()`

Reply:
I rather not change this since I have this `if else` clause which still needs `wp_die()` for `else` part of the statement.

Reply:
I second Konrad here, we should be using WP functions whenever possible ... 
maybe put the result in a variable and use wp_send_json outside the if?

Reply:
Done.